products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -292,6 +293,58 @@ class CAPIBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
   void cancel() override {
     BuildSystemFrontendDelegate::cancel();
   }
+
+  virtual void cycleDetected(const std::vector<core::Rule*>& items) override {
+    llb_build_key_t* rules = (llb_build_key_t*)malloc(items.size() * sizeof(llb_build_key_t));
+    int idx = 0;
+
+    for (std::vector<core::Rule*>::const_iterator it = items.begin(); it != items.end(); ++it) {
+      core::Rule* rule = *it;
+      auto key = BuildKey::fromData(rule->key);
+      llb_build_key_t buildKey;
+      buildKey.key = (char *)malloc(256 * sizeof(char));
```

Comment:
We shouldn't hard code a limit here... it should be easy enough to just make the buildKey.key a copy of the complete key (actually this should simplify the code, you can just use `::strdup`)